quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -122,48 +123,25 @@ ConfigValue getConfigValue(String name, ConfigSourceInterceptorContext context)
         // try to obtain the value for the property we want to map first
         ConfigValue config = convertValue(context.proceed(from));
 
-        if (config == null || config.getValue() == null) {
-            if (mapFrom != null) {
-                // if the property we want to map depends on another one, we use the value from the other property to call the mapper
-                String parentKey = MicroProfileConfigProvider.NS_KEYCLOAK_PREFIX + mapFrom;
-                ConfigValue parentValue = convertValue(context.proceed(parentKey));
-
-                if (parentValue == null) {
-                    // parent value not explicitly set, try to resolve the default value set to the parent property
-                    PropertyMapper<?> parentMapper = PropertyMappers.getMapper(parentKey);
-
-                    if (parentMapper != null && parentMapper.getDefaultValue().isPresent()) {
-                        parentValue = ConfigValue.builder().withValue(Option.getDefaultValueString(parentMapper.getDefaultValue().get())).build();
-                    }
-                }
-
-                return transformValue(name, ofNullable(parentValue == null ? null : parentValue.getValue()), context, null);
-            }
-
-            ConfigValue defaultValue = transformValue(name, this.option.getDefaultValue().map(Option::getDefaultValueString), context, null);
-
-            if (defaultValue != null) {
-                return defaultValue;
-            }
-
-            // now tries any defaults from quarkus
-            ConfigValue current = context.proceed(name);
-
-            if (current != null) {
-                return transformValue(name, ofNullable(current.getValue()), context, current.getConfigSourceName());
-            }
-
-            return current;
+        boolean parentValue = false;
+        if ((config == null || config.getValue() == null) && mapFrom != null) {
+            // if the property we want to map depends on another one, we use the value from the other property to call the mapper
+            config = Configuration.getKcConfigValue(mapFrom);
+            parentValue = true;
+        } 
+
+        if (config != null && config.getValue() != null) {
```

Comment:
> Better, right?

No, you still have to retest if the value is present after obtaining the parent value. If you want to avoid the unnecessary double test case, we can move the mapFrom != null check to the front.

Reply:
@shawkins Ahhh, you're right. Sorry, missed this one. I always try to avoid some comprehensive conditions and have them as much readable as possible. 

But this one is pretty basic, so I'm ok with it.